include/pybind11/pytypes.h

```diff
@@ -2349,7 +2357,11 @@ args_proxy object_api<D>::operator*() const {
 template <typename D>
 template <typename T>
 bool object_api<D>::contains(T &&item) const {
-    return attr("__contains__")(std::forward<T>(item)).template cast<bool>();
+    auto ret = attr("__contains__")(std::forward<T>(item)).template cast<bool>();
+    if (PyErr_Occurred()) {
```

Comment:
Can this actually happen?

I was thinking the line before will either succeed or there will be an exception (`error_already_set`, `cast_error`) in flight already.

Reply:
@rwgk Good catch, my test designed to test this pathway fired fine without the if statement locally.